plugins/package-configuration-providers/dos/src/main/kotlin/DosPackageConfigurationProvider.kt

```diff
@@ -0,0 +1,164 @@
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.clients.dos.DosClient
+import org.ossreviewtoolkit.clients.dos.DosService
+import org.ossreviewtoolkit.clients.dos.PackageConfigurationResponseBody
+import org.ossreviewtoolkit.model.ArtifactProvenance
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.RepositoryProvenance
+import org.ossreviewtoolkit.model.config.LicenseFindingCuration
+import org.ossreviewtoolkit.model.config.LicenseFindingCurationReason
+import org.ossreviewtoolkit.model.config.PackageConfiguration
+import org.ossreviewtoolkit.model.config.PathExclude
+import org.ossreviewtoolkit.model.config.PathExcludeReason
+import org.ossreviewtoolkit.model.config.VcsMatcher
+import org.ossreviewtoolkit.model.utils.PackageConfigurationProvider
+import org.ossreviewtoolkit.model.utils.associateLicensesWithExceptions
+import org.ossreviewtoolkit.model.utils.toPurl
+import org.ossreviewtoolkit.model.utils.toPurlExtras
+import org.ossreviewtoolkit.plugins.packageconfigurationproviders.api.PackageConfigurationProviderFactory
+import org.ossreviewtoolkit.utils.common.Options
+import org.ossreviewtoolkit.utils.spdx.SpdxExpression
+
+data class DosPackageConfigurationProviderConfig(
+    /** The URL where the DOS backend is running. */
+    val url: String,
+
+    /** The secret token to use with the DOS backend. */
+    val token: String,
+
+    /** The timeout for communicating with the DOS backend, in seconds. */
+    val timeout: Long?
+)
+
+open class DosPackageConfigurationProviderFactory :
+    PackageConfigurationProviderFactory<DosPackageConfigurationProviderConfig> {
+    override val type = "DOS"
+
+    override fun create(config: DosPackageConfigurationProviderConfig) = DosPackageConfigurationProvider(config)
+
+    override fun parseConfig(options: Options, secrets: Options) =
+        DosPackageConfigurationProviderConfig(
+            url = options.getValue("url"),
+            timeout = options["timeout"]?.toLongOrNull(),
+            token = secrets.getValue("token")
+        )
+}
+
+/**
+ * A [PackageConfigurationProvider] that loads [PackageConfiguration]s from a Double Open Server instance.
+ */
+class DosPackageConfigurationProvider(config: DosPackageConfigurationProviderConfig) : PackageConfigurationProvider {
+    private val service = DosService.create(
+        config.url,
+        config.token,
+        config.timeout?.let { Duration.ofSeconds(it) }
+    )
+
+    private val client = DosClient(service)
+
+    override fun getPackageConfigurations(packageId: Identifier, provenance: Provenance): List<PackageConfiguration> {
+        val purl = packageId.toPurl(provenance.toPurlExtras())
+
+        val packageResults = runBlocking { client.getPackageConfiguration(purl) }
+        if (packageResults == null) {
+            logger.info { "Unable to find package configuration for '$purl'." }
```

Comment:
Is this case a successful execution with an empty result, or an unsuccessful execution ?

Reply:
It's the latter. As the current log message is misleading, and the `client.getPackageConfiguration(purl)` call already logs, I've simply removed the log completely here.